.gitlab-ci.yml

```diff
@@ -200,3 +201,30 @@ package-oci:
   stage: deploy
   variables:
     PRODUCT_NAME: auto_inject-dotnet
+
+benchmark-serverless:
+  stage: benchmarks
+  image: registry.ddbuild.io/ci/serverless-tools:1
+  tags: ["arch:amd64"]
+  when: on_success
+  needs:
+    - benchmark-serverless-trigger
+  script:
+    - git clone https://gitlab-ci-token:${CI_JOB_TOKEN}@gitlab.ddbuild.io/DataDog/serverless-tools.git ./serverless-tools && cd ./serverless-tools
+    - ./ci/check_trigger_status.sh
+
+benchmark-serverless-trigger:
+  stage: benchmarks
+  needs: []
+  trigger:
+    project: DataDog/serverless-tools
+    strategy: depend
+    branch: jordan.gonzalez/benchmark-dd-trace-dotnet
+  allow_failure: true
```

Comment:
I'm not very familiar with the gitlab syntax, what does this mean in terms of when and what is running here? 🤔 I assume we really want to run these on every _serverless_ PR and on master? Where are the results going? If they're not surfaced in  PR, will anyone look at them? (hint: unlikely, people barely look at them when they _are_ in the PR 😉)

Reply:
`allow_failure: true` indicates that the following jobs should always run, no matter if this failed.

Since this is a trigger job, the `gitsync/codesync` internal service does not display it as a check, hence the creation of another job which later checks if it passed or fail – to notify to the engineer in any PR: `benchmark-serverless` job.

> I assume we really want to run these on every serverless PR and on master?

Correct, this will run on every PR.

> Where are the results going?

As of now, they are being logged in the Serverless Datadog organization, and they are displayed in a dashboard for it. I'll add more details to the PR later.

> If they're not surfaced in PR, will anyone look at them?

As of now, if any PR merged to `master` affects performance in any way, the engineer and serverless will be both notified about it through Slack. We are still looking the possibility of blocking PRs with this check, yet it's still a TBD.

Reply:
Updated description of the PR.

Reply:
Thanks for the details! 🙂 

We are currently heavily heading down a road of reducing the quantity and frequency of testing we do, to try to take the load off the CI... 

I also wonder if we have any data about the variability in the expected values? What counts as a "regression"?

Also, I wonder if we should _not_ run this on every PR, unless we've touched some of the serverless components? Otherwise we stand a very strong chance of it becoming noise that people don't pay any attention to if we get a lot of "false positives" (a problem we have in our existing benchmark/throughput tests)

Reply:
> I also wonder if we have any data about the variability in the expected values? What counts as a "regression"?

As of now, we have set the ceiling to be the the historical cold start + 100ms, if the current build is above that threshold, then we mark as a regression. Although this values are hardcoded, they are very easy to change and don't need a PR in any tracer, just on our downstream services!

> Also, I wonder if we should not run this on every PR, unless we've touched some of the serverless components?

Not really, we want to make sure that every piece of the tracer is checked, for example, if someone adds more components to the Profiler, since we don't support that as of now, that won't affect the Serverless product, the stripping done in our build takes care of that. Yet, if let's say, we add a new binary for ASM (not being stripped since it's new) we want to make sure we strip on our library – it if it's not available for Serverless.

If this becomes flaky, I'm always happy to take responsibility and make sure I can address how to avoid tons of false-positives!